### PR TITLE
Native Ad Banner/MREC formate.

### PR DESCRIPTION
**Pull request by ManojBudumuru-vungle**
This will port the implementation https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/commit/0f05236b5fd917a39676b2d097f1e48752b493ae#

IOS-5823

### CODE REVIEW

File: Vungle/VungleAdapter/ALVungleMediationAdapter.m [hunk 1]
```diff
@@ -435,8 +459,15 @@ - (void)loadAdViewAdForParameters:(id<MAAdapterResponseParameters>)parameters ad
     
     [self updateUserPrivacySettingsForParameters: parameters consentDialogState: self.sdk.configuration.consentDialogState];
     
-    self.bannerRouter = [[ALVungleMediationAdViewAdapterRouter alloc] initVungleAdViewAdDelegate: delegate parentAdapter: self parameters: parameters adFormat: adFormat];
-    [self.bannerRouter loadAdView: placementIdentifier];
+    if ( isNative )
+    {
+        self.nativeAdViewDelegate = [[ALVungleMediationAdapterNativeAdViewDelegate alloc] initVungleNativeAdViewDelegate: delegate parentAdapter: self format:adFormat parameters:parameters];
```
Comment (stanley-vungle) [inline Vungle/VungleAdapter/ALVungleMediationAdapter.m:464]: missing spaces

File: Vungle/VungleAdapter/ALVungleMediationAdapter.m [hunk 2]
```diff
@@ -845,12 +878,220 @@ - (void)requestNativeAd:(NSString *)placementIdentifier
...
     if ( !self.vungleNativeAd )
     {
-        [self.nativeAdDelegate didFailToLoadNativeAdWithError: MAAdapterError.noFill];
+        [self.nativeAdViewDelegate didFailToLoadAdViewAdWithError: MAAdapterError.noFill];
+        return;
+    }
+    
+    if ( ![native.title al_isValidString] )
+    {
+        [self.nativeAdViewDelegate didFailToLoadAdViewAdWithError: [MAAdapterError errorWithCode: -5400 errorString:@"Missing Native Ad Assets"]];
```
Comment (stanley-vungle) [inline Vungle/VungleAdapter/ALVungleMediationAdapter.m:961]: Missing space at errorString

**Review by stanley-vungle — APPROVED** [review]
lgtm. some missing formatting spaces

File: Vungle/VungleAdapter/ALVungleMediationAdapter.m [hunk 3]
```diff
@@ -845,12 +878,220 @@ - (void)requestNativeAd:(NSString *)placementIdentifier
...
+            }
+#pragma clang diagnostic pop
+        }];
+        
+        // Backend will pass down `vertical` as the template to indicate using a vertical native template
+        MANativeAdView *maxNativeAdView;
+        NSString *templateName = [self.parameters.serverParameters al_stringForKey: @"template" defaultValue: @""];
+        if ( [templateName containsString: @"vertical"] )
+        {
+            if ( ALSdk.versionCode < 6140500 )
```
Comment (YueVungle) [inline Vungle/VungleAdapter/ALVungleMediationAdapter.m:906]: I saw `6140500` is used twice, might it could be defined as a constant？Thanks a lot. :)

**Review by YueVungle — APPROVED** [review]
LGTM 👍  except one minor comment

**Review by ashinagawa — APPROVED** [review]
LGTM

**Pull request merged**